File: library/src/main/java/com/raizlabs/android/dbflow/structure/listener/LoadFromCursorListener.java
package com.raizlabs.android.dbflow.structure.listener;

import android.database.Cursor;

/**
 * Description: Marks a {@link com.raizlabs.android.dbflow.structure.Model} as listening to {@link android.database.Cursor}
 * events for custom handling when loading from the DB.
 */
public interface LoadFromCursorListener {

    /**
     * Called when the {@link com.raizlabs.android.dbflow.structure.Model} is loaded from the DB.
     *
     * @param cursor The cursor that is loaded.
     */
    void onLoadFromCursor(Cursor cursor);
}

File: dbflow/src/main/java/com/raizlabs/dbflow5/query/LoadFromCursorListener.kt
package com.raizlabs.dbflow5.query

import android.database.Cursor

import com.raizlabs.dbflow5.structure.Model

/**
 * Description: Marks a [Model] as listening to [Cursor]
 * events for custom handling when loading from the DB.
 */
interface LoadFromCursorListener {

    /**
     * Called when the [Model] is loaded from the DB.
     *
     * @param cursor The cursor that is loaded.
     */
    fun onLoadFromCursor(cursor: Cursor)
}
